x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -230,109 +242,158 @@ export class CloudPlugin implements Plugin<CloudSetup> {
+    });
 
-    // Keep this import async so that we do not load any FullStory code into the browser when it is disabled.
-    const fullStoryChunkPromise = import('./fullstory');
-    const userIdPromise: Promise<string | undefined> = security
-      ? loadFullStoryUserId({ getCurrentUser: security.authc.getCurrentUser })
-      : Promise.resolve(undefined);
-
-    // We need to call FS.identify synchronously after FullStory is initialized, so we must load the user upfront
-    const [{ initializeFullStory }, userId] = await Promise.all([
-      fullStoryChunkPromise,
-      userIdPromise,
-    ]);
-
-    const { fullStory, sha256 } = initializeFullStory({
-      basePath,
-      orgId: fsOrgId,
-      packageInfo: this.initializerContext.env.packageInfo,
+    analytics.registerContextProvider({
+      name: 'cloud_org_id',
+      context$: of({ esOrgId }),
+      schema: {
+        esOrgId: {
+          type: 'keyword',
+          _meta: { description: 'The Cloud Organization ID', optional: true },
+        },
+      },
     });
 
-    // Very defensive try/catch to avoid any UnhandledPromiseRejections
-    try {
-      // This needs to be called syncronously to be sure that we populate the user ID soon enough to make sessions merging
-      // across domains work
-      if (userId) {
-        // Join the cloud org id and the user to create a truly unique user id.
-        // The hashing here is to keep it at clear as possible in our source code that we do not send literal user IDs
-        const hashedId = sha256(esOrgId ? `${esOrgId}:${userId}` : `${userId}`);
-
-        executionContextPromise
-          ?.then(async (executionContext) => {
-            this.appSubscription = executionContext.context$.subscribe((context) => {
-              const { name, page, id } = context;
-              // Update the current context every time it changes
-              fullStory.setVars(
-                'page',
-                omitBy(
-                  {
-                    // Read about the special pageName property
-                    // https://help.fullstory.com/hc/en-us/articles/1500004101581-FS-setVars-API-Sending-custom-page-data-to-FullStory
-                    pageName: `${compact([name, page]).join(':')}`,
-                    app_id_str: name ?? 'unknown',
-                    page_str: page,
-                    ent_id_str: id,
-                  },
-                  isUndefined
-                )
-              );
-            });
+    // This needs to be called synchronously to be sure that we populate the user ID soon enough to make sessions merging
+    // across domains work
+    if (security) {
+      analytics.registerContextProvider({
+        name: 'cloud_user_id',
+        context$: from(loadUserId({ getCurrentUser: security.authc.getCurrentUser })).pipe(
+          filter((userId): userId is string => Boolean(userId)),
+          exhaustMap(async (userId) => {
+            const { sha256 } = await import('js-sha256');
```

Comment:
Import the hashing library only if we got a userId.

Reply:
We can use the browser native digest function here:
https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest

Though it only works on https which might be an OK tradeoff?

Alternatively, wouldn't t be better to send a request to a custom route on the kibana server to get this in a GET/POST request instead of loading an external library?